pkg/spanner/migration.go

```diff
@@ -171,29 +173,46 @@ func dmlToStatements(filename string, data []byte) ([]string, error) {
 
 func inspectStatementsKind(statements []string) (statementKind, error) {
```

Comment:
nit, The current code looks unnecessarily complicated to me. How about refactoring it like below?

```go
func inspectStatementsKind(statements []string) (statementKind, error) {
	var hasDDL, hasDML, hasPartitionedDML bool
	for _, s := range statements {
		switch {
		case isDML(s):
			hasDML = true
		case isPartitionedDML(s):
			hasPartitionedDML = true
		default:
			hasDDL = true
		}
	}

	switch {
	case hasDDL && !hasDML && !hasPartitionedDML:
		return statementKindDDL, nil
	case !hasDDL && hasDML && !hasPartitionedDML:
		return statementKindDML, nil
	case !hasDDL && !hasDML && hasPartitionedDML:
		return statementKindPartitionedDML, nil
	default:
		return "", errors.New("DDL, DML (INSERT), and partitioned DML (UPDATE or DELETE) must not be combined in the same migration file")
	}
}
```